Fix status command

HeDo88TH opened the pull request:
Closes #208 
Simple refactor and changed behaviour of the function `checkUpdate`: it now skips mtime check for folders.

pierotofy commented on src/cmd/status.cpp:
```diff
@@ -2,8 +2,10 @@
-#include <iostream>
 #include "status.h"
+
+#include <iostream>
```
Can we please disable auto-code formatting? :)

HeDo88TH replied:
The formatter just replaced tabs with spaces and swapped a couple of includes.

pierotofy commented:
Ok I can't discern what was done by the auto-formatter and what was actually changed; please re-open another PR this time disabling the auto-formatting changes.

HeDo88TH commented:
> Ok I can't discern what was done by the auto-formatter and what was actually changed; please re-open another PR this time disabling the auto-formatting changes.

The auto formatter just replaced tabs with spaces and sorted the includes.

Here are the actual code changes:

- Added `NotModified` enum value to `FileStatus`
- Moved `checkUpdate` function declaration from `dbops.h` to `status.h`
- `checkUpdate` function now returns a `FileStatus` and checks internally if the file/folder exists
- `checkUpdate` function skips `mtime` check if it's a folder

pierotofy commented:
> please re-open another PR this time disabling the auto-formatting changes.

HeDo88TH commented:
@pierotofy reverted all auto format changes. 
Soon or later we should clean up the codebase because many files have different coding styles, spacing and indentation conventions. I will open a PR to do this.

pierotofy reviewed (changes requested):
Thanks. :+1: 

This still needs some work.

When you change a function's signature, search the codebase where it's used; you will find that it's being called in:

 - `dbops.cpp:287`, `dbops.cpp:486` and in both places, it breaks.

```
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $ ddb sync
U	a
[piero:/data/test/test] $
```

pierotofy commented:
Attaching below the checklist for PRs:

 - [ ] Compiles on Windows and Linux
 - [ ] Manual tests on both Windows and Linux to confirm the functionality/command works
 - [ ] Manual tests on commands/functionality that might have been affected by these changes.
 - [ ] There are no styling changes to unrelated code.

pierotofy commented:
Thanks! :+1:  Looks good.
